hermes-client/src/main/java/pl/allegro/tech/hermes/client/metrics/DropwizardMetricsProvider.java

```diff
@@ -0,0 +1,40 @@
+package pl.allegro.tech.hermes.client.metrics;
+
+import com.codahale.metrics.MetricRegistry;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class DropwizardMetricsProvider implements MetricsProvider {
+
+    private final MetricRegistry metrics;
+
+    public DropwizardMetricsProvider(MetricRegistry metrics) {
+        this.metrics = metrics;
+    }
+
+    @Override
+    public void counterIncrement(String name) {
+        metrics.counter(name).inc();
+    }
+
+    @Override
+    public void counterIncrement(String prefix, String name, Map<String, String> tags) {
+        counterIncrement(buildCounterName(prefix, name, tags));
+    }
+
+    @Override
+    public void timerRecord(String name, long duration, TimeUnit unit) {
+        metrics.timer(name).update(duration, unit);
+    }
+
+    @Override
+    public void histogramUpdate(String name, int value) {
+        metrics.histogram(name).update(value);
+    }
+
+    private String buildCounterName(String prefix, String name, Map<String, String> tags) {
+        return prefix + "." + name + "." + tags.values().stream().collect(Collectors.joining("."));
```

Comment:
How about `String.join(".", tags.values())` instead of `tags.values().stream().collect(Collectors.joining(".")`